methods.py

```diff
@@ -863,12 +863,16 @@ def clean_cache(cache_path: str, cache_limit: int, verbose: bool):
     texts = []
     stats = []
     for file in files:
+        # Save file stats to rewrite after modifying.
+        tmp_stat = os.stat(file)
```

Comment:
The `OSError` catchblock was meant to catch instances where accessing the file itself failed. This should be in an equivalent block, though the code could be refactored to account for catching multiple OS issues at once:
```py
    for file in files:
        # Save file stats to rewrite after modifying.
        try:
            tmp_stat = os.stat(file)
            # Failing a utf-8 decode is the easiest way to determine if a file is binary.
            try:
                with open(file, encoding="utf-8") as out:
                    out.read(1024)
            except UnicodeDecodeError:
                stats.append((file, *tmp_stat[6:8]))
                # Restore file stats after reading.
                os.utime(file, (tmp_stat[7], tmp_stat[8]))
            else:
                texts.append(file)
        except OSError:
            print_error(f'Failed to access cache file "{file}"; skipping.')
```

Reply:
Good catch, will fix